conda_build/config.py

```diff
@@ -101,7 +101,7 @@ def ext(self):
 no_rewrite_stdout_env_default = "false"
 ignore_verify_codes_default = []
 exit_on_verify_error_default = False
-conda_pkg_format_default = CondaPkgFormat.V1
+conda_pkg_format_default = CondaPkgFormat.V2
```

Comment:
This is the main change. Everything else is to fix assumptions about `tar.bz2` being always default.